studies/calgames_kinematics/src/main/java/org/team100/Kinematics.java

```diff
@@ -1,24 +1,101 @@
+    private double m_elbowAngle;
+    private double m_shoulderAngle;
+    private double m_zLength;
 
-    public Kinematics(double armLength, double wristLength) {
+    public Kinematics(double armLength, double manipulatorLength) {
         m_armLength = armLength;
-        m_wristLength = wristLength;
+        m_manipulatorLength = manipulatorLength;
     }
 
     public Pose2d forward(Config config) {
         // forward kinematics
-        return new Pose2d();
+        System.out.println("TESTING FORWARD KINEMATICS: \n" );
+        System.out.println("PRESETS:" );
+
+        System.out.println("Arm Length: " + m_armLength);
+        System.out.println("Manipulator Length: " + m_manipulatorLength);
+        System.out.println("Shoulder Height: " + config.m_shoulderHeight());
+        System.out.println("Shoulder Angle: " + config.m_shoulderAngle());
+        System.out.println("Elbow Angle: " + config.m_elbowAngle());
+
+        double armXcomp  = (m_armLength * Math.cos(Math.toRadians(config.m_shoulderAngle())));
+        double manipulatorXcomp = (m_manipulatorLength * Math.sin(Math.toRadians(config.m_elbowAngle()-(180-90-config.m_shoulderAngle()))));
+        double xComp = (armXcomp)+ (manipulatorXcomp);
+
+
+        double armYcomp = (m_armLength * Math.sin(Math.toRadians(config.m_shoulderAngle())));
+        double manipulatorYcomp = (m_manipulatorLength * Math.cos(Math.toRadians(config.m_elbowAngle()-(180-90-config.m_shoulderAngle()))));
+        double yComp =  (armYcomp) - (manipulatorYcomp) + config.m_shoulderHeight(); //subtraction becuse the manipulator is always pointing down?
+
+        double manipulatorRotation = Math.toRadians(180-90-(config.m_elbowAngle()-(180-90-config.m_shoulderAngle())));
+
+        System.out.println("\nOUTPUTS:");
+        System.out.println("armXcomp: " + armXcomp);
+        System.out.println("armYcomp: " + armYcomp);
+        System.out.println("manipulatorXcomp: " + manipulatorXcomp);
+        System.out.println("manipulatorYcomp: " + manipulatorYcomp);
+        System.out.println("X Point: " + (xComp));
+        System.out.println("Y Point: " + yComp);
+        System.out.println("Manipulator Rotation: " + (manipulatorRotation));
+
+        return new Pose2d(xComp, yComp, new Rotation2d(manipulatorRotation));
     }
 
     public Config inverse(Pose2d pose) {
         // inverse kinematics
-        return new Config(0, 0, 0);
+
+        System.out.println("\nTESTING INVERSE KINEMATICS:\n" );
+
+        System.out.println("PRESETS:" );
+        System.out.println("Arm Length: " + m_armLength);
+        System.out.println("Manipulator Length: " + m_manipulatorLength);
+        System.out.println("Goal X: " + pose.getX());
+        System.out.println("Goal Y: " + pose.getY());
+        System.out.println("Goal Rotation: " + pose.getRotation().getDegrees());
+
+        //1. find the elbow position from the manipulator length and the goal 2dpose. ('p' in my notes) - KYM (CORRECT - 9/13/2025)
```

Comment:
it's great to reference notes like this !  great!  you should also take a picture of the notes, stick the pic in a doc, make sure the doc is in the team drive, and link to the doc here, so that anyone reading this comment can see the diagram.  you could also write more notes in that same doc -- like a "design doc" for your code.